### ISSUE TITLE
2.17.2 Set Sticky Bit on All World-Writable Directories fails is no files found

### ISSUE THREAD

**Issue by memelet** [awailly/cis-ubuntu-ansible#101]
If section 2.17 is skipped section 2.17.2 will result in the following error:

```
TASK: [cis | 2.17.2 Set Sticky Bit on All World-Writable Directories (Scored)] ***
fatal: [10.0.139.156] => with_items expects a list or a set
fatal: [10.0.136.223] => with_items expects a list or a set
```

This is really a general problem with ansible (at least with 1.9) in that even though a task is skipped its with_items is still evaluated.

**Comment by memelet** [comment]
If acceptable I will submit a PR that defaults all with_items variables with default([])... ?

**Comment by awailly** [comment]
Is the PR #102 fixing all of the `with_items` issues? I will look, but any way to fix it in the role parameter? Otherwise it will be error-prone if we need to add more steps and forget about this issue.

**Comment by awailly** [comment]
Ok so the issue is http://stackoverflow.com/questions/35470667/ansible-with-items-when-item-is-defined. This is specified in the documentation at http://docs.ansible.com/ansible/playbooks_conditionals.html#the-when-statement. I was completely unaware of this, and I think this is a strong design issue from Ansible. Anyway, we need to merge #102 with some modifications ASAP to prevent the problem. I will describe the modifications in the #102, but mainly applying this only to commands using both `with_items` and `when`.

Duplicate in #89.

**Comment by pchaigno** [comment]
Fixed with #108.
